### PR TITLE
feat: support for stylelint 14 and added missing rules

### PR DESCRIPTION
**Pull request by BeeMargarida**
* `Stylelint v14` droped automatic inferral of syntax and no longer includes syntax for SCSS and does not extract styles from HTML and others: https://stylelint.io/migration-guide/to-14/
   * This requires the use of a config for vue `stylelint-config-html/vue` (requires stylelint-config-html and postcss-html) and for scss `stylelint-config-standard-scss`
* `stylelint-config-standard` added new rules in `v23` that are against the standards used: https://github.com/stylelint/stylelint-config-standard/releases

### CODE REVIEW

**Comment by BeeMargarida** [comment]
@joamag There were some missing changes needed for working with stylelint v14, which added several breaking changes.

File: index.js [hunk 1]
```diff
@@ -18,6 +22,16 @@ module.exports = {
         "no-empty-source": null,
         "no-descending-specificity": null,
         "rule-empty-line-before": "always",
+        "shorthand-property-no-redundant-values": null,
```
Comment (joamag) [inline index.js]: I think the keys are sorted alphabetically :)

**Review by joamag — CHANGES_REQUESTED** [review]

**Comment by joamag** [comment]
@BeeMargarida It's critical to avoid changes to the current way of styling CSS and SCSS so every time we find something breaking the consistency let's try to find alternatives

**Comment by BeeMargarida** [comment]
> @BeeMargarida It's critical to avoid changes to the current way of styling CSS and SCSS so every time we find something breaking the consistency let's try to find alternatives

@joamag These changes are required so that the way of styling CSS and SCSS does not change in our current projects using stylelint v14. The only thing that needed change (in ripe-white, for example), was the way we import/declare font names: https://github.com/ripe-tech/ripe-white/pull/936/commits/562c1c303c32f3572e077ab5b82fb21a88d39cae
We were using `url` with and without `"`, which is not allowed by the rules [`function-url-quotes`](https://github.com/stylelint/stylelint/blob/main/lib/rules/function-url-quotes/README.md) and naming fonts without `"` [`font-family-name-quotes`](https://github.com/stylelint/stylelint/blob/main/lib/rules/font-family-name-quotes/README.md).

I also tested in another repo (pulse) and only two changes are needed (which do not follow the standards - use of `'` instead of `"` and use `pascalCase` instead of `kebab-case` in CSS classes.

**Review by joamag — APPROVED** [review]

**Comment by BeeMargarida** [comment]
@joamag Would it be possible to create a new version with these changes?

**Pull request merged**